src/cow-react/modules/swap/hooks/useIsEthFlow.ts

```diff
@@ -1,7 +1,9 @@
 import { useDetectNativeToken } from '@cow/modules/swap/hooks/useDetectNativeToken'
+import { getEthFlowEnabled } from '@cow/modules/swap/helpers/getEthFlowEnabled'
 
 export function useIsEthFlow(): boolean {
   const { isNativeIn, isWrapOrUnwrap } = useDetectNativeToken()
+  const isEnabled = getEthFlowEnabled()
 
-  return isNativeIn && !isWrapOrUnwrap
+  return isEnabled && isNativeIn && !isWrapOrUnwrap
```

Comment:
localStorage flag takes precedence when deciding if we are doing the native eth flow